Sources/SlidingContainerViewController.swift

```diff
@@ -0,0 +1,400 @@
+
+    @IBOutlet open weak var navView: UIView?
+
+    public weak var presenter: SlidingContainerPresenter!
+
+    public weak var delegate: SlidingContainerDelegate!
+
+    public var upperController: ContainedViewController!
+
+    public var lowerController: ContainedViewController!
+
+    public let upperView: UIView = UIView()
+
+    public let lowerView: UIView = UIView()
+
+    var panDirection: Int = 0
+
+    private var initialPositionSetUp = false
+
+    private var isDragging = false
+
+    private var lastChange = 0.0
+
+    open override func viewDidLoad() {
+        super.viewDidLoad()
+
+        upperController = presenter.upperController(for: self)
+        lowerController = presenter.lowerController(for: self)
+
+        addChildViewController(upperController.viewController)
+        addChildViewController(lowerController.viewController)
+
+        navView = navView == nil ? view : navView
+
+        let dView = UIView(frame: CGRect(x: 0, y: 0, width: 0, height: 42))
+        dView.backgroundColor = .darkGray
+        dragView = dragView == nil ? dView : dragView
+
+        navView?.addSubview(upperView)
+        navView?.addSubview(dragView!)
+        navView?.addSubview(lowerView)
+
+        upperView.addSubview(upperController.viewController.view)
+        lowerView.addSubview(lowerController.viewController.view)
+
+        lowerController.didMinimizeControllerWith(diff: getMovementDifference())
+        upperController.didMaximizeControllerWith(diff: getMovementDifference())
+
+        lowerController.viewController.didMove(toParentViewController: self)
+        upperController.viewController.didMove(toParentViewController: self)
+
+        let panGesture = UIPanGestureRecognizer(
+            target: self,
+            action: #selector(self.panDragView(sender:))
+        )
+        panGesture.delegate = self
+        panGesture.delaysTouchesBegan = false
+        panGesture.delaysTouchesEnded = false
+        panGesture.cancelsTouchesInView = false
+        navView?.addGestureRecognizer(panGesture)
+    }
+
+    open override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        if !initialPositionSetUp {
+            initialPositionSetUp = true
+            drawViews()
+        }
+        lowerController.viewController.view.frame = frameForLowerController()
+        upperController.viewController.view.frame = frameForUpperController()
+    }
+
+    func getMovementDifference() -> CGFloat {
+        guard let navView = navView else { return 0 }
+        let diff = navView.frame.height - presenter.upperViewMin(for: self) - presenter.lowerViewMin(for: self) - (dragView?.frame.height ?? 42)
+        return diff
+    }
+
+    // MARK: Frame Management
+
+    func drawViews() {
+        guard let navView = navView else { return }
+        let navHeight = navView.frame.height
+        let navWidth = navView.frame.width
+        let dragHeight = (dragView?.frame.height ?? 42)
```

Comment:
maybe you can move `42` to a constant or all this `dragView?.frame.height ?? 42` to a property, what do you think?